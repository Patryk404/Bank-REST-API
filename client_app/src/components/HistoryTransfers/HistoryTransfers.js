import React from 'react';
import classes from './HistoryTransfers.module.css';
import HistoryTransfer from './HistoryTransfer/HistoryTransfer';
import Aux from '../../hoc/Auxiliary/Auxiliary';
const historyTransfers = (props)=>{
    let transfers;
    if (props.history)
    {
        transfers = props.history.map(
            (element,index)=>{// setting list with appropriate key
                return (
                <HistoryTransfer key={index}>{element}</HistoryTransfer>
                );
            }
        )
    }
    else { 
        transfers = null;
    }
    return(
        <Aux>
            <div className={classes.Transfers}>
                <h1 style={{fontSize: '20px',fontWeight: 'normal'}}>History of Transfers</h1>
                    <ul>
                        {transfers}
                    </ul>
            </div>
        </Aux>
    );
}; 

export default historyTransfers;