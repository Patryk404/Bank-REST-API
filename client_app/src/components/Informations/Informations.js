import React from 'react';
import classes from './Informations.module.css';
const informations = (props)=>{
    return(
        <div className={classes.Informations}>
            <h1>Informations</h1>
            <ul>
            <li><h2 style={{fontSize: '36px'}}>{props.user.name} {props.user.surname}</h2></li>
            <li><h3>Cash: {props.user.cash}</h3></li>
            <li><h3>Bill Number: {props.user.bill}</h3></li>
            </ul>
        </div>
    );
}

export default informations;