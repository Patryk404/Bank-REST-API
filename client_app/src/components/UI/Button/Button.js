import React from 'react';
import classes from './Button.module.css';
const button = (props)=>{
    let style = [];
    style.push(classes.Button);
    if (props.styled==='red')
    {
        style.push(classes.Red);
    }
    return(
    <button className={style.join(' ')} onClick={props.click}>{props.children}</button>
    );
}

export default button;