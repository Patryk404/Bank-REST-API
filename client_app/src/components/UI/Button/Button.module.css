.Button {
    color: #fff !important;
    text-transform: uppercase;
    text-decoration: none;
    background: #60a3bc;
    padding: 20px;
    margin: 10px;
    border-radius: 50px;
    display: inline-block;
    border: none;
    cursor: pointer;
    transition: all 0.4s ease 0s;
    width: 20%;
}

.Red {
    background: red;
    border-radius: 10px;
}

.Button:hover { 
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
    -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
    transition: all 0.4s ease 0s;
}

@media (max-width: 600px){
    .Button {
        display: block;
        margin: 20px auto;
        width: 80%;
        text-align: center;
    }
}