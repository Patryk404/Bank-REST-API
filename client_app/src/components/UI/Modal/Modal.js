import React from 'react';
import classes from './Modal.module.css';
import Backdrop from '../Backdrop/Backdrop';
import Aux from '../../../hoc/Auxiliary/Auxiliary';
const modal = (props)=> {
    return (
        <Aux>
            <Backdrop show={props.show} click={props.clickonbackdrop}/>
            <div className={classes.Modal}
                style={{
                    transform: props.show ? 'translateY(0)' : 'translateY(-100vh)',//animtion if our props show is true or not 
                    opacity: props.show ? '1' : '0' // visibility 
               }}>
               {props.children}
            </div>
        </Aux>
    );
}

export default modal;