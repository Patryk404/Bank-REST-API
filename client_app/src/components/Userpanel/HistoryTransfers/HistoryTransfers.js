import React from 'react';
import classes from './HistoryTransfers.module.css';
import HistoryTransfer from './HistoryTransfer/HistoryTransfer';
import Aux from '../../../hoc/Auxiliary/Auxiliary';
const historyTransfers = (props)=>{
    const transfers = props.history.map(
        (element,index)=>{// setting list with appropriate key
            return (
            <HistoryTransfer key={index}>{element}</HistoryTransfer>
            );
        }
    )
    return(
        <Aux>
            <div className={classes.Transfers}>
                <h1 style={{fontSize: '20px',fontWeight: 'normal'}}>History of Transfers</h1>
                    <ul>
                        {transfers}
                    </ul>
            </div>
        </Aux>
    );
}; 

export default historyTransfers;