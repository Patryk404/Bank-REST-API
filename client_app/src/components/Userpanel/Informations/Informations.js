import React from 'react';
import classes from './Informations.module.css';
const informations = (props)=>{
    return(
        <div className={classes.Informations}>
            <h1>Informations</h1>
            <ul>
            <li><h2>{props.name_and_surname}</h2></li>
            <li><h3>Cash: {props.cash}</h3></li>
            </ul>
        </div>
    );
}

export default informations;