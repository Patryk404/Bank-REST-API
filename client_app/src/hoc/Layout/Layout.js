import React from 'react';
import Logo from '../../components/Logo/Logo';
import Aux from '../Auxiliary/Auxiliary';
const layout = (props)=>{
    return(
        <Aux>
            <Logo />
            {props.children}
        </Aux>
    );
}; 

export default layout;