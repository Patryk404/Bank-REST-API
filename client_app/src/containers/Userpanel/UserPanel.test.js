import UserPanel from './UserPanel';
import HistoryTransfers from '../../components/HistoryTransfers/HistoryTransfers';
import React from 'react';
import Aux from '../../hoc/Auxiliary/Auxiliary';
import Layout from '../../hoc/Layout/Layout';
import Informations from '../../components/Informations/Informations';
import MakeTransfer from '../../components/Maketransfer/Maketransfer';

import {configure,shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Modal from '../../components/UI/Modal/Modal';
import Button from '../../components/UI/Button/Button';

configure({adapter: new Adapter});

describe('<UserPanel />',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<UserPanel/>);
        localStorage.setItem('token','6845976957468945');
    })

    it('matches snapshot',()=>{
        wrapper = renderer.create(<UserPanel/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render <HistoryTransfers> component',()=>{
        wrapper.setState({loggedUser:{
            transfers:[]
        }});
        expect(wrapper.find(HistoryTransfers)).toHaveLength(1);
    });
    it('should render modal with MakeTransfer if we click into button ',()=>{
        expect(wrapper.state().showMakeTransfer).toEqual(false);
        const ButtonComponent = wrapper.find('button').at(1); //selecting second node
        ButtonComponent.props().click();
        //console.log('find button component', ButtonComponent.debug());
        //wrapper.instance().makeTransferButtonHandler();
        expect(wrapper.find(Modal)).toHaveLength(1);
    })
    it('should render a make_transfer modal if we click on button',()=>{
        const button = wrapper.find(Button).at(1).children();
        const our_button_component = wrapper.find(Button).at(1);
        expect(button.text()).toEqual('Make Transfer');
        our_button_component.simulate('click');
        expect(wrapper.find(MakeTransfer)).toHaveLength(1);
    })
    it('should render <Aux> component',()=>{
        expect(wrapper.find(Aux)).toHaveLength(1);
    })
    it('should render <Layout> component',()=>{
        expect(wrapper.find(Layout)).toHaveLength(1);
    })
    it('should render <Informations> component',()=>{
        expect(wrapper.find(Informations)).toHaveLength(1);
    })
    it('should render <MakeTransfer> component if stateshowMakeTransfer is true',()=>
    {
        wrapper.setState({showMakeTransfer: true});
        expect(wrapper.find(MakeTransfer)).toHaveLength(1);
    })
    it('should render h1 if we are not authenticated',()=>{
        localStorage.removeItem('token');
        expect(wrapper.find('h1')).toHaveLength(1);
    })
    
})