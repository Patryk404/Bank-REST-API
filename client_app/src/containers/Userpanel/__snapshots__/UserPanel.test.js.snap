// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserPanel /> matches snapshot 1`] = `
<div
  className="Panel"
>
  <div
    className="Logo_logged"
  >
    <img
      alt="Logo"
      src="logo.png"
    />
  </div>
  <h1
    style={
      Object {
        "fontWeight": "normal",
      }
    }
  >
    Panel
  </h1>
  <div>
    <button
      className="Button Red"
      onClick={[Function]}
    >
      Log Out
    </button>
    <button
      className="Button"
      onClick={[Function]}
    >
      Make Transfer
    </button>
  </div>
  <div
    className="Transfers"
  >
    <h1
      style={
        Object {
          "fontSize": "20px",
          "fontWeight": "normal",
        }
      }
    >
      History of Transfers
    </h1>
    <ul />
  </div>
  <div
    className="Informations"
  >
    <h1>
      Informations
    </h1>
    <ul>
      <li>
        <h2
          style={
            Object {
              "fontSize": "36px",
            }
          }
        >
          
           
          
        </h2>
      </li>
      <li>
        <h3>
          Cash: 
        </h3>
      </li>
      <li>
        <h3>
          Bill Number: 
          
        </h3>
      </li>
    </ul>
  </div>
</div>
`;
